Unable to install on mac

We have a user with a problem getting `serialport` installed on his Mac OSx 10.10.2.

See: https://community.spark.io/t/tutorial-spark-cli-on-mac-osx-25-july-2014/5225/70 and https://community.spark.io/t/tutorial-spark-cli-on-mac-osx-25-july-2014/5225/85

Can someone help figure out what might be causing the issue? Thanks!

Same problem here. Slightly different message. 

```
gyp verb build dir attempting to create "build" dir: /Users/thomas/work-perso/node-serialport/build
gyp verb build dir "build" dir needed to be created? /Users/thomas/work-perso/node-serialport/build
gyp verb build/config.gypi creating config file
gyp verb build/config.gypi writing out config file: /Users/thomas/work-perso/node-serialport/build/config.gypi
gyp verb config.gypi checking for gypi file: /Users/thomas/work-perso/node-serialport/config.gypi
gyp verb common.gypi checking for gypi file: /Users/thomas/work-perso/node-serialport/common.gypi
gyp verb gyp gyp format was not specified; forcing "make"
gyp info spawn python
gyp info spawn args [ '/usr/local/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'make',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/thomas/work-perso/node-serialport/build/config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/usr/local/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/thomas/.node-gyp/0.12.3/common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=/Users/thomas/.node-gyp/0.12.3',
gyp info spawn args   '-Dmodule_root_dir=/Users/thomas/work-perso/node-serialport',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.' ]
gyp info ok
gyp info it worked if it ends with ok
gyp verb cli [ '/usr/local/bin/node',
gyp verb cli   '/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js',
gyp verb cli   'build',
gyp verb cli   '--fallback-to-build',
gyp verb cli   '--module=/Users/thomas/work-perso/node-serialport/build/serialport/v1.7.0/Release/node-v14-darwin-x64/serialport.node',
gyp verb cli   '--module_name=serialport',
gyp verb cli   '--module_path=/Users/thomas/work-perso/node-serialport/build/serialport/v1.7.0/Release/node-v14-darwin-x64' ]
gyp info using node-gyp@1.0.3
gyp info using node@0.12.3 | darwin | x64
gyp verb command build []
gyp verb build type Release
gyp verb architecture x64
gyp verb node dev dir /Users/thomas/.node-gyp/0.12.3
gyp verb `which` succeeded for `make` /usr/bin/make
gyp info spawn make
gyp info spawn args [ 'V=1', 'BUILDTYPE=Release', '-C', 'build' ]
  c++ '-D_DARWIN_USE_64_BIT_INODE=1' '-D_LARGEFILE_SOURCE' '-D_FILE_OFFSET_BITS=64' '-DBUILDING_NODE_EXTENSION' -I/Users/thomas/.node-gyp/0.12.3/src -I/Users/thomas/.node-gyp/0.12.3/deps/uv/include -I/Users/thomas/.node-gyp/0.12.3/deps/v8/include -I../node_modules/nan  -Os -gdwarf-2 -mmacosx-version-min=10.5 -arch x86_64 -Wall -Wendif-labels -W -Wno-unused-parameter -fno-rtti -fno-exceptions -fno-threadsafe-statics -fno-strict-aliasing -MMD -MF ./Release/.deps/Release/obj.target/serialport/src/serialport.o.d.raw  -c -o Release/obj.target/serialport/src/serialport.o ../src/serialport.cpp
  c++ '-D_DARWIN_USE_64_BIT_INODE=1' '-D_LARGEFILE_SOURCE' '-D_FILE_OFFSET_BITS=64' '-DBUILDING_NODE_EXTENSION' -I/Users/thomas/.node-gyp/0.12.3/src -I/Users/thomas/.node-gyp/0.12.3/deps/uv/include -I/Users/thomas/.node-gyp/0.12.3/deps/v8/include -I../node_modules/nan  -Os -gdwarf-2 -mmacosx-version-min=10.5 -arch x86_64 -Wall -Wendif-labels -W -Wno-unused-parameter -fno-rtti -fno-exceptions -fno-threadsafe-statics -fno-strict-aliasing -MMD -MF ./Release/.deps/Release/obj.target/serialport/src/serialport_unix.o.d.raw  -c -o Release/obj.target/serialport/src/serialport_unix.o ../src/serialport_unix.cpp
../src/serialport_unix.cpp:180:7: error: redefinition of 'fd'
  int fd = open(data->path, flags);
      ^
../src/serialport_unix.cpp:156:15: note: previous definition is here
int setup(int fd, OpenBaton *data) {
              ^
../src/serialport_unix.cpp:191:5: error: non-void function 'setup' should return a value [-Wreturn-type]
    return;
    ^
../src/serialport_unix.cpp:703:20: warning: comparison of array 'device.locationId' not equal to a null pointer is always true [-Wtautological-pointer-compare]
        if (device.locationId != NULL) {
            ~~~~~~~^~~~~~~~~~    ~~~~
../src/serialport_unix.cpp:706:20: warning: comparison of array 'device.vendorId' not equal to a null pointer is always true [-Wtautological-pointer-compare]
        if (device.vendorId != NULL) {
            ~~~~~~~^~~~~~~~    ~~~~
../src/serialport_unix.cpp:709:20: warning: comparison of array 'device.productId' not equal to a null pointer is always true [-Wtautological-pointer-compare]
        if (device.productId != NULL) {
            ~~~~~~~^~~~~~~~~    ~~~~
../src/serialport_unix.cpp:712:20: warning: comparison of array 'device.manufacturer' not equal to a null pointer is always true
      [-Wtautological-pointer-compare]
        if (device.manufacturer != NULL) {
            ~~~~~~~^~~~~~~~~~~~    ~~~~
../src/serialport_unix.cpp:715:20: warning: comparison of array 'device.serialNumber' not equal to a null pointer is always true
      [-Wtautological-pointer-compare]
        if (device.serialNumber != NULL) {
            ~~~~~~~^~~~~~~~~~~~    ~~~~
5 warnings and 2 errors generated.
make: *** [Release/obj.target/serialport/src/serialport_unix.o] Error 1
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:269:23)
gyp ERR! stack     at ChildProcess.emit (events.js:110:17)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:1074:12)
gyp ERR! System Darwin 14.3.0
gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "build" "--fallback-to-build" "--module=/Users/thomas/work-perso/node-serialport/build/serialport/v1.7.0/Release/node-v14-darwin-x64/serialport.node" "--module_name=serialport" "--module_path=/Users/thomas/work-perso/node-serialport/build/serialport/v1.7.0/Release/node-v14-darwin-x64"
gyp ERR! cwd /Users/thomas/work-perso/node-serialport
gyp ERR! node -v v0.12.3
gyp ERR! node-gyp -v v1.0.3
gyp ERR! not ok
node-pre-gyp ERR! build error
node-pre-gyp ERR! stack Error: Failed to execute '/usr/local/bin/node /usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js build --fallback-to-build --module=/Users/thomas/work-perso/node-serialport/build/serialport/v1.7.0/Release/node-v14-darwin-x64/serialport.node --module_name=serialport --module_path=/Users/thomas/work-perso/node-serialport/build/serialport/v1.7.0/Release/node-v14-darwin-x64' (1)
node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (/Users/thomas/work-perso/node-serialport/node_modules/node-pre-gyp/lib/util/compile.js:83:29)
node-pre-gyp ERR! stack     at ChildProcess.emit (events.js:110:17)
node-pre-gyp ERR! stack     at maybeClose (child_process.js:1015:16)
node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:1087:5)
node-pre-gyp ERR! System Darwin 14.3.0
node-pre-gyp ERR! command "node" "/Users/thomas/work-perso/node-serialport/node_modules/.bin/node-pre-gyp" "install" "--fallback-to-build"
node-pre-gyp ERR! cwd /Users/thomas/work-perso/node-serialport
node-pre-gyp ERR! node -v v0.12.3
node-pre-gyp ERR! node-pre-gyp -v v0.6.7
node-pre-gyp ERR! not ok
Failed to execute '/usr/local/bin/node /usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js build --fallback-to-build --module=/Users/thomas/work-perso/node-serialport/build/serialport/v1.7.0/Release/node-v14-darwin-x64/serialport.node --module_name=serialport --module_path=/Users/thomas/work-perso/node-serialport/build/serialport/v1.7.0/Release/node-v14-darwin-x64' (1)
```

--- a/src/serialport_unix.cpp
+++ b/src/serialport_unix.cpp
@@ -177,7 +177,7 @@ int setup(int fd, OpenBaton *data) {
   if(data->hupcl == false) {
     flags &= ~HUPCL;
   }
-  int fd = open(data->path, flags);
+  fd = open(data->path, flags);
 
   if (fd == -1) {
     snprintf(data->errorString, sizeof(data->errorString), "Cannot open %s", data->path);
@@ -188,7 +188,7 @@ int setup(int fd, OpenBaton *data) {
   int cloexec = fcntl(fd, F_SETFD, FD_CLOEXEC);
   if (cloexec == -1) {
     snprintf(data->errorString, sizeof(data->errorString), "Cannot open %s", data->path);
-    return;
+    return -1;
   }
 
   // struct sigaction saio;

